clang/lib/3C/DeclRewriter.cpp

```diff
@@ -487,11 +552,42 @@ void DeclRewriter::doDeclRewrite(SourceRange &SR, DeclReplacement *N) {
   }
 
   rewriteSourceRange(R, SR, Replacement);
+
+  SourceLocation L = getLocationAfter(N->getDecl()->getEndLoc(),
+                                      A.getSourceManager(), A.getLangOpts());
+  emitSupplementaryDeclarations(N->getSupplementaryDecls(), L);
```

Comment:
It looks like you're assuming that `L` will always point to the comma or semicolon after the multi-decl member.  Apparently this is not true if there is a space before the comma or semicolon.  Example:
```c
#include <stdlib.h>

void test4() {
  int *a = malloc(10*sizeof(int)) , b;
  a++;
}
```
On the `range_bounds` branch as-is, `3c -alltypes` gives me an assertion failure; I haven't investigated why.  On the `range_bounds.matt_rewrite_offset` branch, we can see the supplementary declaration being inserted in the wrong place, causing a syntax error:
```c
#include <stdlib.h>

void test4() {
  _Array_ptr<int> __3c_tmp_a : count(10) = malloc<int>(10*sizeof(int))
_Array_ptr<int> a : bounds(__3c_tmp_a, __3c_tmp_a + 10) = __3c_tmp_a;
 ;
int b;
  a++;
}
```
I presume this is why we used `getNextComma` in `rewriteMultiDecl`.

In general, it seems fragile to me to compute one source location here and one in `rewriteMultiDecl` and hope they match up.  I think it would be more robust and easier to understand if we call `emitSupplementaryDeclarations` directly from `rewriteMultiDecl` with the location computed there.  I prototyped this approach in [the `range_bounds.getnextcomma` branch](https://github.com/correctcomputation/checkedc-clang/compare/range_bounds.matt_rewrite_offset...range_bounds.getnextcomma), which is based on my previous `range_bounds.matt_rewrite_offset` branch.  This passes the regression tests; I haven't tried the benchmark tests.  The need to conditionalize the `getNextCommaOrSemicolon` to work around an assertion failure in an existing regression test is very unfortunate, but if I tried to come up with a real fix to the deeper problem with `getNextCommaOrSemicolon`, that would require a bunch more regression tests that the fix is correct, and I don't think any of us wants to deal with that now. :frowning_face:  With the `range_bounds.getnextcomma` branch as it is, I think our general testing philosophy would only call for adding my example above as a regression test.